runtime/lua/vim/treesitter.lua

```diff
@@ -466,4 +466,74 @@ function M.foldexpr(lnum)
   return M._fold.foldexpr(lnum)
 end
 
+---@param parser vim.treesitter.LanguageTree
+---@return {group:string,range:integer[]}[]
+function M.get_text_highlight(parser)
```

Comment:
`parser` is a `vim.treesitter.LanguageTree`, which is already a complex object. Should this instead be a method on that object?

Reply:
Ok